examples-new/webhooks/server.py

```diff
@@ -0,0 +1,157 @@
+        To do so, run this command:
+
+            pip install -r requirements.txt
+    """)
+    print(message, file=sys.stderr)
+    sys.exit(1)
+
+# This example uses Flask, a micro web framework written in Python.
+# For more information, check out the documentation: http://flask.pocoo.org
+# Create a Flask app, and load the configuration file.
+app = Flask(__name__)
+app.config.from_json('config.json')
+
+# Check for dummy configuration values.
+# If you are building your own application based on this example,
+# you can remove this check from your code.
+cfg_needs_replacing = [
+    key for key, value in app.config.items()
+    if isinstance(value, str) and value.startswith("replace me")
+]
+if cfg_needs_replacing:
+    message = textwrap.dedent("""
+        This example will only work if you replace the fake configuration
+        values in `config.json` with real configuration values.
+        The following config values need to be replaced:
+        {keys}
+        Consult the README.md file in this directory for more information.
+    """).format(keys=", ".join(cfg_needs_replacing))
+    print(message, file=sys.stderr)
+    sys.exit(1)
+
+# Teach Flask how to find out that it's behind an ngrok proxy
+app.wsgi_app = ProxyFix(app.wsgi_app)
+
+
+@app.route('/webhook', methods=['GET', 'POST'])
+def webhook():
+    """
+    When the Flask server gets a request at the `/webhook` URL, it will run
+    this function. Most of the time, that request will be a genuine webhook
+    notification from Nylas. However, it's possible that the request could
+    be a fake notification from someone else, trying to fool our app. This
+    function needs to verify that the webhook is genuine!
+    """
+    # When you first tell Nylas about your webhook, it will test that webhook
+    # URL with a GET request to make sure that it responds correctly.
+    # We just need to return the `challenge` parameter to indicate that this
+    # is a valid webhook URL.
+    if request.method == "GET" and "challenge" in request.args:
+        print(" * Nylas connected to the webhook!")
+        return request.args["challenge"]
+
+    # Alright, this is a POST request, which means it's a webhook notification.
+    # The question is, is it genuine or fake? Check the signature to find out.
+    is_genuine = verify_signature(
+        message=request.data,
+        key=app.config["NYLAS_OAUTH_CLIENT_SECRET"].encode('utf8'),
+        signature=request.headers.get('X-Nylas-Signature'),
+    )
+    if not is_genuine:
+        return "Signature verification failed!", 401
+
+    # Alright, we have a genuine webhook notification from Nylas!
+    # Let's find out what it says...
+    data = request.get_json()
+    for delta in data["deltas"]:
+        # This is the part of the code where you would process the information
+        # from the webhook notification. Each delta is one change that happened,
+        # and might require fetching message IDs, updating your database,
```

Comment:
Can you mention that any processing or fetching should be done **async** please? We recommend that customers queue work and return a 200 to prevent Nylas POST from timing out. I believe our timeout is 5 minutes right now.